report.go

```diff
@@ -23,23 +23,36 @@ type Report struct {
 	SendEmail       bool       `json:"send_email,omitempty"`
 }
 
+// CreateReportOptions is used to specify columns included in a report.
+type CreateReportOptions struct {
+	Columns           []string `json:"columns,omitempty"`
+	AdditionalColumns []string `json:"additional_columns,omitempty"`
+}
+
+type createReportRequest struct {
+	*CreateReportOptions
+	Report *Report `json:"report,omitempty"`
+}
+
 // CreateReport generates a new report. Valid Fields for input are StartDate,
 // EndDate and SendEmail. A new Report object is returned. Once the Status is
-// available, the report can be downloded from the provided URL for 30 seconds.
+// available, the report can be downloaded from the provided URL for 30 seconds.
 //	c := easypost.New(MyEasyPostAPIKey)
 //	c.CreateReport(
 //		"payment_log",
 //		&easypost.Report{StartDate: "2016-10-01", EndDate: "2016-10-31"},
 //	)
-func (c *Client) CreateReport(typ string, in *Report) (out *Report, err error) {
-	err = c.post(context.Background(), "reports/"+typ, in, &out)
+func (c *Client) CreateReport(typ string, in *Report, opts *CreateReportOptions) (out *Report, err error) {
+	req := &createReportRequest{CreateReportOptions: opts, Report: in}
+	err = c.post(context.Background(), "reports/"+typ, req, &out)
 	return
 }
 
 // CreateReportWithContext performs the same operation as CreateReport, but
 // allows specifying a context that can interrupt the request.
-func (c *Client) CreateReportWithContext(ctx context.Context, typ string, in *Report) (out *Report, err error) {
-	err = c.post(ctx, "reports/"+typ, in, &out)
+func (c *Client) CreateReportWithContext(ctx context.Context, typ string, in *Report, opts *CreateReportOptions) (out *Report, err error) {
```

Comment:
`typ` should be `type`, I know you didn't do it but let's fix it.

Reply:
`type` is a reserved word in Golang

Reply:
Le sigh. Fine